### ISSUE TITLE
Why another git detector/dumper ?

### ISSUE THREAD

**Issue by noraj** [WangYihang/GitHacker#1]
Hi,

Can you add a **Why another git detector/dumper"** section in the README? This is an usual practice for developer that created a new tool that looks very similar to existing tool, to explain why their tool if different/needed, what new it is bringing that other tool can't offer due to technical dept, why it was not possible to join current maintainer of existing tool to help them adding a need feature rather than creating a new fork or tool from scratch that just divide the community.

There is already well-known quality tools such as [dvcs-ripper](https://github.com/kost/dvcs-ripper) or [GitTools](https://github.com/internetwache/GitTools) so you must have a good reason to develop yet another tool from scratch and alone.
Because I don't see why people should use this tool rather than existing one that are more maintained. Both dvcs-ripper and GitTools can dump git directory and not only dump the last revision like the githack tool you mentioned.

**Comment by WangYihang** [comment]
yeah, this is a good question, I will make a video to compare these tools and explain why this tools is better than others on git folder leakage. :D

**Comment by WangYihang** [comment]
@noraj Updated, please see README.md
Here are the results:
![result](https://camo.githubusercontent.com/0522b5934e8fdd75aa67e63d2288f6f13efedae5/68747470733a2f2f75706c6f61642d696d616765732e6a69616e7368752e696f2f75706c6f61645f696d616765732f323335353037372d363462646531626366363137653063662e706e673f696d6167654d6f6772322f6175746f2d6f7269656e742f7374726970253743696d61676556696577322f322f772f31323430)

**Comment by noraj** [comment]
@WangYihang Looks good! what is the meaning of the environment column?

**Comment by WangYihang** [comment]
That is a configuration of webserver
`Indexes` means webserver will list files of an folder when there is no index.html file
`-Indexes` means webserver will return forbidden in that situation

**Comment by WangYihang** [comment]
You can just see the videos in the readme file, I recorded everything when I was doing these testing

**Comment by arthaud** [comment]
You should also compare the ability to find packed refs. This is important to dump big repositories.

Looking at the code, I believe your tool doesn't handle this. [dvcs-ripper](https://github.com/kost/dvcs-ripper) and my [own tool](https://github.com/arthaud/git-dumper) both handle it well.

**Comment by WangYihang** [comment]
@arthaud Cool, a nice resource to learn. I am preparing the examination of Chinese postgraduate entrance this year, so I will add the capability next year. thank you so much for your suggestion!
